src/main/java/com/jmb/springfactory/service/task/TaskServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+  }
+
+  @Override
+  public Class<? extends TaskDto> getDtoClazz() {
+    return TaskDto.class;
+  }
+
+  @Override
+  public Class<? extends BusinessObjectBase> getBoClazz() {
+    return BusinessObjectBase.class;
+  }
+  
+  @Override
+  public void update(TaskDto taskDto, Integer idTask) throws ServiceLayerException {
+    taskValidatorService.validateOnUpdate(taskDto);
+    super.update(taskDto, idTask);
+  }
+  
+  @Override
+  public TaskDto save(Integer orderId, TaskDto taskDto) throws ServiceLayerException, NotFoundException, 
+    PersistenceLayerException {
+    
+    addInitialInformation(orderId, taskDto);
+    return super.save(taskDto);
+  }
+  
+  @Override 
+  public void delete(Integer id) {
+    
+    final Consumer<Task> updateAsDeleted = task -> {
+       task.setStatus(TaskStatusEnum.DELETED); 
+       try {
+         taskMySQLService.save(task);
+       } catch (Exception e) {
+         serviceLog.error(String.format("A error has ocurred when update as DELETED the task [%s]", e.getMessage()));
+      }
+    };
+
+    serviceLog.info(String.format("Search task with id %s", id));
+    taskMySQLService.findOne(id).ifPresent(updateAsDeleted);
+  }
+
+  /**
+   * Add essential information to create a task
+   * @param orderId
+   * @param taskDto
+   * @throws NotFoundException
+   * @throws PersistenceLayerException
+   */
+  private void addInitialInformation(Integer orderId, TaskDto taskDto)
+      throws NotFoundException, PersistenceLayerException {
+
+    serviceLog.info(String.format("Adding task into order %s", orderId));
+    taskDto.setOrder(productionOrderService.findOne(orderId));
+    
+    if (notExist(taskDto.getOrderNumber())) {
+      serviceLog.info("The order number is not defined");
+      taskDto.setOrderNumber(getLastPositionOrder(orderId));
+    }
+
+    serviceLog.info(String.format("Set task initial status as %s", TASK_INITIAL_STATUS));
+    taskDto.setStatus(TASK_INITIAL_STATUS);
+  }
+
+  /**
+   * Get the last position order of all task of specific order
+   * @param orderId
+   * @return
+   * @throws PersistenceLayerException
+   */
+  private Integer getLastPositionOrder(Integer orderId) throws PersistenceLayerException {
+
+    final Integer numberOrders = taskMySQLService.countByOrderId(orderId).intValue();
+    final Integer lastNumberOrder = numberOrders + 1;
+
+    serviceLog.info(String.format("There are %s task for order %s, the order of the new one is %s", 
+        orderId, numberOrders, lastNumberOrder));
+
+    return lastNumberOrder;
+  }
+
+  @Override
+  public List<TaskDto> findAll(QueryTaskObject queryParams) {
+    serviceLog.info(String.format("Search task with the follow arguments: %s", queryParams.toString()));
+    return taskMySQLService.findAll(queryParams).map(this::entityToDto).collect(Collectors.toList());
+  }
+  
+  @SuppressWarnings("unchecked")
+  @Override
+  public void addComment(Integer idTask, CommentDto comment) throws NotFoundException, ServiceLayerException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)